### PR TITLE
Installation instructions for Linux with systemd

### PR DESCRIPTION
**Pull request by jorgebay**
Provide installation and upgrade instructions for Linux.

### CODE REVIEW

File: docs/LINUX.md [hunk 1]
```diff
@@ -26,97 +26,97 @@ gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.
...
+systemctl status logdna-agent
+```
+
+## Upgrading from LogDNA Agent v1/v2 for Linux
+
+LogDNA Agent v1/v2 for Linux used initd to start as a daemon and `.conf` files to define the settings.
+
+To upgrade, make sure you use the new repository on `https://assets.logdna.com` and run the upgrade command
+of your package manager:
+
```
Comment (tanberry) [inline docs/LINUX.md:77]: "... run the appropriate upgrade command for your specific package manager."

File: docs/LINUX.md [hunk 2]
```diff
@@ -26,97 +26,97 @@ gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.
...
+gpgcheck=1
+gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.repo
+
+# Update the package
+sudo yum update logdna-agent
+```
+
+The agent will uninstall the previous version and reuse the existing configuration file, by default
+located in `/etc/logdna.conf`.
+
```
Comment (tanberry) [inline docs/LINUX.md:108]: Just a nit, but is "reuse" the right word, or does it "rewrite" or "replace"....? What technically happens?

File: docs/LINUX.md [hunk 3]
```diff
@@ -26,97 +26,97 @@ gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.
...
+If you defined a configuration file on a different location, you can specify it on your systemd unit file:
+
```
Comment (tanberry) [inline docs/LINUX.md:110]: Is this rewrite accurate? I wanted to be clear that it means you can specify a diff conf file, and just ignore the one we gave them.

 "If you defined a `logdna-agent.conf` configuration file on a different location, and want to use it instead, you can specify it..."

File: docs/LINUX.md [hunk 4]
```diff
@@ -26,97 +26,97 @@ gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.
...
+```unit file (systemd)
+[Service]
+Environment="LOGDNA_CONFIG_FILE=/your/path/to/logdna.conf"
+```
 
-You can use `--help` flag to list all the command and environment variable options. For example with agent
-version `3.3.0`:
+Followed by a restart:
```
Comment (tanberry) [inline docs/LINUX.md:117]: "Next, restart the agent:"

**Review by tanberry — APPROVED** [review]

**Comment by tanberry** [comment]
Thanks @jorgebay for writing the Linux install docs! I am excited to try these steps out later today.

File: docs/LINUX.md [hunk 5]
```diff
@@ -26,97 +26,97 @@ gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.
...
+environment variable:
+
+```unit file (systemd)
+[Service]
+Environment="LOGDNA_INGESTION_KEY=<YOUR INGESTION KEY HERE>"
+```
+
+You can see all the available environment variables using `logdna-agent --help`. For example, you can set the tags to
+attach to each line using `LOGDNA_TAGS` variable:
+
```
[the remark this answers is not in the document]
Comment (jorgebay) [reply docs/LINUX.md:52]: I'll separate it into two 2 paragraphs.

> Though to be honest, I am not sure what that last sentence means.

It's supposed to be similar to the existing instructions, where other optional settings are shown (look for `You can configure the agent to tag` on the [existing instructions](https://docs.logdna.com/docs/logdna-agent-1)).

File: docs/LINUX.md [hunk 6]
```diff
@@ -26,97 +26,97 @@ gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.
...
+```unit file (systemd)
+[Service]
+Environment="LOGDNA_INGESTION_KEY=<YOUR INGESTION KEY HERE>"
+Environment="LOGDNA_TAGS=production"
+```
+
+After saving the configuration, you can start the service:
```
[the remark this answers is not in the document]
Comment (jorgebay) [reply docs/LINUX.md:60]: Makes sense, I'll add it.

File: docs/LINUX.md [hunk 7]
```diff
@@ -26,97 +26,97 @@ gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.
...
 
-The LogDNA Agent exposes short argument abbreviations for commonly used options (`-k`, `-t`, `-d` and `-c`).
+You can check the status of the agent using `systemctl status`:
+
+```shell script
+systemctl status logdna-agent
+```
+
+## Upgrading from LogDNA Agent v1/v2 for Linux
+
```
[the remark this answers is not in the document]
Comment (jorgebay) [reply docs/LINUX.md:72]: afaik, we didn't mention that the previous agent was node.js based, it's an implementation detail from the user pov.

On [this doc](https://docs.logdna.com/docs/logdna-agent-1) it's referred as agent v1 for linux.

File: docs/LINUX.md (hunk 2 above)
Comment (jorgebay) [reply docs/LINUX.md:108]: Technically, it uses the conf file as-is, the new agent is able to read to the previous agent's format.

**Comment by jorgebay** [comment]
I've adapted the doc based on your suggestions @tanberry 

I've tested it on as a non-root user and I've added the `sudo` commands when needed.

File: docs/LINUX.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (tanberry) [reply docs/LINUX.md:74]: I like how you have it now, thanks, agree about leaving Node out of it, thanks!

File: docs/LINUX.md [hunk 8]
```diff
@@ -26,97 +26,97 @@ gpgkey=https://assets.logdna.com/logdna.gpg" | sudo tee /etc/yum.repos.d/logdna.
...
 ```
 
-# Usage
+## Usage
 
-You can start sending logs by setting the [ingestion key][ingestion-key]:
+The agent uses [systemd](https://systemd.io/) to run as a Linux daemon.
+
+After installing the package, you should enable it on systemd and set the ingestion key:
+
```
[the remark this answers is not in the document]
Comment (tanberry) [reply docs/LINUX.md:34]: Maybe I am missing it but did you add the name of the package? Is it the systemd package that needs to be installed, in sentence above?

File: docs/LINUX.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (tanberry) [reply docs/LINUX.md:54]: I think I need to learn more about this... I'll ask in Slack, but leet's go ahead with how you have it worded.. ;-)

**Review by tanberry — APPROVED** [review]
Thanks @Jorge for adding the comments and helping me better understand some of the sections.

**Pull request merged**